account_payment_order_invoice_cancel/models/account_invoice.py

```diff
@@ -0,0 +1,50 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, models, _
+
+
+class AccountInvoice(models.Model):
+
+    _inherit = 'account.invoice'
+
+    @api.multi
+    def action_cancel(self):
+        invoices_with_payment_orders = self.\
+            _get_invoice_linked_with_payment_orders()
+        self = self - invoices_with_payment_orders
+        res = True
+        if self:
+            res = super(AccountInvoice, self).action_cancel()
```

Comment:
wouldn't it be better to call the wizard for all invoices if at least one invoice is linked with a payment line ?
If we imagine a process that allows to cancel several invoices, those that are not associated with a line of payment will be automatically canceled but a wizard will open for the others. If you click on the cancel button of this wizard, the others are already cancel. This may be disturbing.
What do you think about that ?

Reply:
I understand, but the wizard warns about the fact that invoices are linked to a payment order. I think that if we show invoices which are not linked to a payment order, it would be disturbing too.